app/types/factory.py

```diff
@@ -0,0 +1,49 @@
+from abc import ABC, abstractmethod
+
+from sqlalchemy.ext.asyncio import AsyncSession
+
+
+class Factory(ABC):
+    def __init__(
+        self,
+        name: str,
```

Comment:
Is the name of the factory really useful? Could we directly use the name of the class?